lib/enum.ml

```diff
@@ -1,5 +1,16 @@
 open Ppxlib
 
+module Attr = struct
+  let value =
+    Attribute.declare
+      "ppx_enum.enum.value"
+      Attribute.Context.constructor_declaration
+      Ast_pattern.(single_expr_payload (estring __))
```

Comment:
Does the `__` (double underscore) mean anything special?

Reply:
Yes - it means "match anything here and capture it". In all honesty, the AST pattern stuff is the most confusing bit of ppxlib that we found, so this might not be the most optimal way to have done things (but the ppxlib docs are currently under heavy improvement).

See https://github.com/ocaml-ppx/ppxlib/blob/f199d7eb29b6eb12c9ef0585727996ddc13bf9a7/src/ast_pattern.mli#L91 for more details